libcaf_net/caf/net/stream_transport.hpp

```diff
@@ -155,6 +159,7 @@ class stream_transport {
       CAF_LOG_ERROR("send_buffer_size: " << socket_buf_size.error());
       return std::move(socket_buf_size.error());
     }
+    owner->register_reading();
```

Comment:
Why should the transport unconditionally decide that the application *has* to read something right away?

Reply:
The lifetime of `socket_manager` instances is bound to them being registered for reading or writing. As soon as a `socket_manager` is not registered for either of both, it is dereferenced by the multiplexer and hence, removed. Managers are only registered for writing, when they actually have to write data which leaves registering any manager for reading to keep them from being removed right away.